core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java

```diff
@@ -68,27 +71,26 @@ public HadoopTables(Configuration conf) {
    */
   @Override
   public Table load(String location) {
-    TableOperations ops = newTableOps(location);
-    if (ops.current() == null) {
+    //Possibly Load a Metadata Table
```

Comment:
How about parse out the metadata table name, if present?

Reply:
I made a different implementation here where we will instead do

```
Name, Type = tryParseMetadataLocation

if (name, type).{
  loadMetadata
} else {
  loadBasic
}
```